### PR TITLE
Show patients with national health id when searching with from scan screen

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/3218/show-patients-with-national-health-id-when-searching-with-from-scan-screen

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt [hunk 1]
```diff
@@ -57,17 +57,20 @@ class ScanSimpleIdUpdate @Inject constructor(
...
     } else {
       val patientId = patients.first().uuid
       OpenPatientSummary(patientId)
     }
   }
 
+  private fun multiplePatientsWithId(identifier: Identifier): ScanSimpleIdEffect {
+    return when (identifier.type) {
+      BpPassport -> OpenShortCodeSearch(BpPassport.shortCode(identifier))
+      else -> OpenPatientSearch(null, initialSearchQuery = identifier.value)
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt:70]: Why are we passing the identifier as null here and in the tests as well? Does this mean apart from BP passport just search with ID regardless of whatever identifier it is?

File: app/src/test/java/org/simple/clinic/scanid/ScanSimpleIdUpdateTest.kt [hunk 2]
```diff
@@ -162,4 +162,18 @@ class ScanSimpleIdUpdateTest {
             hasEffects(OpenPatientSearch(null, "12345612345612"))
         ))
   }
+
+  @Test
+  fun `when entered code is not a short code, then open patient search`() {
+    val shortCode = "12345678901234"
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/scanid/ScanSimpleIdUpdateTest.kt:168]: ~Can we rename `shortCode` to `enteredCode` here?~ I have a renaming PR coming up for this package will include this as well.

**Review by Honey14 — APPROVED** [review]
Looks good.

File: app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt:70]: Yep, if we pass an identifier to instant search it tries to bring up the sheet that asks if we want to register or assign it to a patient. In this case, this is not a blank identifier, we know this is already assigned to a patient. I should probably switch to using factory constructors for this to make it much more apparent what it's doing 🤔

**Pull request merged**